### DISCUSSION TITLE
Add in-game skin library & owned beatmaps list

### DISCUSSION THREAD

**Discussion by auroraongithub** [ppy/osu#29049] [Ideas] [answered]
An in-game skin library would be neat for ease of access, especially for mobile players who still have to download the file and use a file explorer like ZArchive to open the .osk file. Would be very useful for pc players as well. For the skins that get uploaded to the catalog, you can either let anyone upload a skin with tags (including r18 ones), have a team manually approve and tag every skin uploaded, or have a team of skin curators, basically "approved by osu!" skins.

For the owned beatmaps list, its more of a quality of life addition. Lets say you bought another laptop/pc, probably because you broke your old one so you can't redownload exactly what maps you had, this will make a file/list that contains every single map you've ever owned (removes deleted ones and doesn't include maps not in the osu! website) and store it somewhere online for the client to give them the option to download when they log in their account from a new pc. Or maybe store it all somewhere in a database linked to your account, idk.

**Comment by Joehuu** [comment, marked as answer]
Skin library is originally tracked in https://github.com/ppy/osu/discussions/17759, don't forget to upvote!

See https://github.com/ppy/osu/discussions/28405 and the linked discussions there to find the answer to your second suggestion.

**Reply by auroraongithub** [reply]
thanks!
